### DISCUSSION TITLE
Generate 2D view of wall

### DISCUSSION THREAD

**Discussion by richardklingler** [IfcOpenShell/IfcOpenShell#5413] [Q&A]
Good afternoon (o;

Just started today to explore ifcopenshell as I need to process IFC models to generate 2D wall drawings to visualize what needs to be installed on a particular wall.

Can that be achieved at all?

thanks for hints (o;
richard

**Comment by aothms** [comment]
Definitely, but there is no automation yet for the wall-specific positioning of the drawing plane.

You could use draw.py as a starting point.

https://github.com/IfcOpenShell/IfcOpenShell/blob/v0.8.0/src/ifcopenshell-python/ifcopenshell/draw.py

We have our own convention within the ifcopenshell project to define a drawing 'camera' as an IfcAnnotation with a camera 'frustrum' defined as an IfcBlock geometry looking towards the negative Z axis of the annotation local placement. (I don't know your background, can imagine this sounds a bit foreign). These annotations objects can be created programmatically.

You can also look at Bonsai for a full scriptable visual environment based on Blender in which the same and more can be accomplished.

**Comment by Moult** [comment]
There's also quite a lot of code in the Bonsai directory that could be generalised into the IfcOpenShell API as parts start to mature (e.g. the camera definitions). Would you be interested in helping refactor this?
